src/common/types/validity_mask.cpp

```diff
@@ -46,6 +46,15 @@ string ValidityMask::ToString(idx_t count) const {
 	return result;
 }
 
+bool ValidityMask::Equal(ValidityMask &mask, idx_t count) const {
+	for (idx_t i = 0; i < count; i++) {
+		if (RowIsValid(i) != mask.RowIsValid(i)) {
```

Comment:
Can you check the validity entries directly? That should be significantly faster than checking using `RowIsValid`, which does a bunch of bitwise ops per value.

Also can use an early out if both masks are not set (`AllValid()`).